--- sara_b.c	2020-04-07 22:01:53.994052589 -0400
+++ sara_a.c	2020-04-07 22:01:06.520719786 -0400
@@ -18,6 +18,9 @@
 /* Xlib */
 #include <X11/Xlib.h>
 #include <X11/Xft/Xft.h>
+#ifdef XINERAMA
+#include <X11/extensions/Xinerama.h>
+#endif
 
 #define BUTTONMASK              	(ButtonPressMask|ButtonReleaseMask)
 #define MOUSEMASK               	(BUTTONMASK|PointerMotionMask)
@@ -1133,6 +1136,15 @@
 	changemon(m, YesFocus);
 }
 
+#ifdef XINERAMA
+static int isuniquegeom(XineramaScreenInfo* unique, size_t n, XineramaScreenInfo* info){
+	while (n--)
+		if (unique[n].x_org == info->x_org && unique[n].y_org == info->y_org)
+			return 0;
+	return 1;
+}
+#endif
+
 /* a la dwm 6.1 */
 void
 updategeom(){
@@ -1140,6 +1152,31 @@
 	client* c;
 	monitor* m, * oldmhead = mhead;
 
+#ifdef XINERAMA
+	if (XineramaIsActive(dis)){
+		int i, j, ns;
+		XineramaScreenInfo* unique;
+		XineramaScreenInfo* info = XineramaQueryScreens(dis, &ns);
+
+      		/* only consider unique geometries as separate screens */
+		unique = ecalloc(ns, sizeof(XineramaScreenInfo));
+		for (i=0, j=0;i < ns;i++)
+			if (isuniquegeom(unique, j, &info[i]))
+				memcpy(&unique[j++], &info[i], sizeof(XineramaScreenInfo));
+		XFree(info);
+		
+		mhead = m = createmon(0, unique[0].x_org, unique[0].y_org,
+				unique[0].width, unique[0].height);
+		for (i=1;i < j;i++){
+			m->next = createmon(i, unique[i].x_org, unique[i].y_org,
+					unique[i].width, unique[i].height);
+			m = m->next;
+		}
+
+		free(unique);
+
+	} else
+#endif
 	{
 		mhead = createmon(0, 0, 0, sw, sh);
 	}
